src/app.ts

```diff
@@ -68,14 +68,13 @@ const cli = new Cli({
             process.exit(1);
         });
 
-        process.on("SIGTERM", async () => {
+        process.on("SIGTERM", () => {
             log.info("Got SIGTERM");
-            try {
-                await main.killBridge();
-            } catch (ex) {
-                log.warn("Failed to kill bridge, exiting anyway");
-            }
-            process.exit(1);
+            main.killBridge().catch((ex) => {
+                log.warn("Failed to kill bridge, exiting anyway", ex);
+            }).finally(() => {
+                process.exit(0);
```

Comment:
We changed the status here, because 1 is not correct if the bridge managed to exit imo.

Reply:
One way you can handle this to give more context is to use a bitmask for possible failure cases, flipping a bit in the mask when you encounter a failure mode during shutdown. Then you take the mask bits, convert it to a number and use that as your exit code. So the process shutting down but having had issues to kill the bridge could end up being a status code of 2, whereas is `killBridge()` didn't fail you'd exit with 0 etc.

Reply:
it feels slightly overkill to bitmask it, given we basically either exit-OK, or exit-something-failed. I can just return a 2 though on failure, reserving 1 for actual crashes.